DNN Platform/Library/Services/Installer/Installers/ScriptInstaller.cs

```diff
@@ -267,18 +317,30 @@ protected override void ProcessFile(InstallFile file, XPathNavigator nav)
                     // This is the initial script when installing
                     this._installScript = file;
                 }
-                else if (file.Name.StartsWith("upgrade.", StringComparison.InvariantCultureIgnoreCase))
+                else if (file.Name.StartsWith("preupgrade.", StringComparison.InvariantCultureIgnoreCase)
+                    || type.Equals("preupgrade", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    this._preUpgradeScript = file;
+                }
+                else if (file.Name.StartsWith("upgrade.", StringComparison.InvariantCultureIgnoreCase)
+                    || type.Equals("postupgrade", StringComparison.InvariantCultureIgnoreCase))
                 {
                     this._upgradeScript = file;
                 }
+                else if (file.Name.StartsWith("schema.", StringComparison.InvariantCultureIgnoreCase)
+                    || type.Equals("schema", StringComparison.InvariantCultureIgnoreCase))
+                {
+                    this._schemaScripts.Add(file);
+                }
                 else if (type.Equals("install", StringComparison.InvariantCultureIgnoreCase))
                 {
                     // These are the Install/Upgrade scripts
                     this.InstallScripts[file.Version] = file;
                 }
-                else
+                else if (file.Name.StartsWith("uninstall.", StringComparison.InvariantCultureIgnoreCase)
+                    || type.Equals("uninstall", StringComparison.InvariantCultureIgnoreCase))
                 {
-                    // These are the Uninstall scripts
+                    // This is the Uninstall script
```

Comment:
It looks like multiple uninstall scripts are supported
```suggestion
                    // These are the Uninstall scripts
```

Reply:
My bad. It is.